pokemon-checker/src/components/PokemonDisplay/PokemonDisplay.tsx

```diff
@@ -27,33 +26,26 @@ export const PokemonDisplay = (props: displayProps) => {
     () => new PokemonFactory(),
     []
   );
+  const { getPokemonData, pokemonUrl } = props;
   /**
    * Create a PokemonObject from the results retrieved
    * @param pokemonRetrieved the pokemon retrieved from the API
    */
   const createPokemonObject = useCallback(
-    (pokemonRetrieved: IPokemonData): void => {
-      let pokemonToDisplay = pokemonFactory.createPokemon(pokemonRetrieved);
-      setPokemonObject(pokemonToDisplay);
+    (url: string) => {
+      if (url && url.length > 0) {
+        getPokemonData(url)
+          .then((data) => pokemonFactory.createPokemon(data))
+          .then((pokemon) => pokemonFactory.fetchAbilities(pokemon))
+          .then((pokemon) => setPokemonObject(pokemon));
```

Comment:
Hot dang this is what I was trying to do in the service

Thanks for clearing that up, I was getting bodied by the `.then` calls after a successful fetch